### ISSUE TITLE
sortier messes with order of function overrides

### ISSUE THREAD

**Issue by martaver** [snowcoders/sortier#942]
# Language
Typescript

# Sample input source code
```
export interface LogFn<TScopes> {
  (key: keyof TScopes): (...args: any[]) => void;
  (...args: any[]): void;
}
```

# Expected output
No changes. The order of function overrides in typescript is significant. e.g. if I declare a logger like so:
```
export interface Log {
  <TScopes extends Record<string, boolean>>(
    enable: boolean,
    name: string,
    scopes: TScopes
  ): LogFn<TScopes>;
}
const log = Log(true, 'MyLogger', {
  myScope: false,
});
```

With the original order, when I specify a 'scope' as an argument in the first function call, typescript knows I am making a scoped logger call, and returns LogFn accordingly. However, if the first argument doesn't match one of the keys of the scopes, then it'll pass the arguments directly to logFn.

E.g.
```
log('myScope')('my message to be logged'); // This should work.
```

However, when sortier switches the two override definitions, it changes the precedence for override discovery and breaks things.

# Actual output (or error message)
```
export interface LogFn<TScopes> {
  (...args: any[]): void;
  (key: keyof TScopes): (...args: any[]) => void;
}
```

Suggest that the order of function overrides in types and interfaces should not be changed :)

**Comment by k2snowman69** [comment]
Released 3.1.3 which contains the fix

**Comment by martaver** [comment]
Great, thanks!
